asyncio/selector_events.py

```
@@ -634,6 +645,11 @@ def __init__(self, loop, sock, protocol, waiter=None,
         self._eof = False
         self._paused = False
 
+        # Disable the Nagle algorithm -- small writes will be
+        # sent without waiting for the TCP ACK.  This generally
+        # decreases the latency (in some cases significantly.)
+        _set_nodelay(self._sock)
```

> Good catch. We need to add functional unittests for TCP/UNIX transports to catch these sort of errors...

Tests probably shows nothing, since error is swallowed. Or, tests should test that TCP_NODELAY has been actually set.